modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Foldable.kt

```diff
@@ -157,9 +161,11 @@ interface Foldable<F> {
    *
    * Similar to foldM, but using a Monoid<B>.
    */
-  fun <G, A, B, TC> Kind<F, A>.foldMapM(tc: TC, f: (A) -> Kind<G, B>): Kind<G, B>
-    where TC : Monad<G>, TC : Monoid<B> = tc.run {
-    foldM(tc, tc.empty()) { b, a -> f(a).map { b.combine(it) } }
+  fun <G, A, B, MA, MO> Kind<F, A>.foldMapM(ma: MA, mo: MO, f: (A) -> Kind<G, B>): Kind<G, B>
+      where MA : Monad<G>, MO : Monoid<B> = ma.run {
```

Comment:
This will crash Android builds with the new compiler. You cannot nest two `run` directly.

Reply:
Fixed here https://github.com/arrow-kt/arrow/pull/1058